app/src/main/java/com/example/parstagram/fragments/ProfileFragment.java

```diff
@@ -0,0 +1,205 @@
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.widget.ShareActionProvider;
+import androidx.core.view.MenuItemCompat;
+import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.GridLayoutManager;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;
+
+import com.bumptech.glide.Glide;
+import com.example.parstagram.LoginActivity;
+import com.example.parstagram.Post;
+import com.example.parstagram.PostsAdapter;
+import com.example.parstagram.ProfilePostsAdapter;
+import com.example.parstagram.R;
+import com.parse.FindCallback;
+import com.parse.ParseException;
+import com.parse.ParseFile;
+import com.parse.ParseQuery;
+import com.parse.ParseUser;
+import com.parse.SaveCallback;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ProfileFragment extends Fragment {
+
+    public static final String TAG = "ProfileFragment";
+    public static final int RESULT_LOAD_IMAGE = 1;
+    private RecyclerView rvPosts;
+    private ProfilePostsAdapter adapter;
+    private List<Post> allPosts;
+    private SwipeRefreshLayout swipeContainer;
+    private Button btnLogout;
+    private ImageView ivProfile;
+    private TextView tvUsername;
+    private Button btnSetProfile;
+
+    public ProfileFragment() {
+        // Required empty public constructor
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.fragment_profile, container, false);
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        rvPosts = view.findViewById(R.id.rvPosts);
+        // Lookup the swipe container view
+        swipeContainer = (SwipeRefreshLayout) view.findViewById(R.id.swipeContainer);
+        btnLogout = (Button) view.findViewById(R.id.btnLogout);
+        ivProfile = (ImageView) view.findViewById(R.id.ivProfile);
+        tvUsername = (TextView) view.findViewById(R.id.tvUsername);
+        btnSetProfile = (Button) view.findViewById(R.id.btnSetProfile);
+
+        allPosts = new ArrayList<>();
+        adapter = new ProfilePostsAdapter(getContext(), allPosts);
+        tvUsername.setText(ParseUser.getCurrentUser().getUsername());
+
+        ParseFile profilePicture = ParseUser.getCurrentUser().getParseFile("profilePicture");
+        if (profilePicture != null) {
+            Glide.with(getContext())
```

Comment:
I didn't do this in mine, but now I realize we did use a lot of Glide functions with the exact same format. Creating a helper function will definitely help shorten our code